### PR TITLE
[MM-32136] add "telemetry: disabled" comments to the Config struct

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We will add `go vet` check to raise a warning if a config is not added to telemetries. In order to ignore some fields, we can use a comment to exclude those from the check.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32137

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: services/telemetry/telemetry.go [hunk 1]
```diff
@@ -768,7 +768,7 @@ func (ts *TelemetryService) trackConfig() {
 		"is_default_global_relay_smtp_username": isDefault(*cfg.MessageExportSettings.GlobalRelaySettings.SmtpUsername, ""),
 		"is_default_global_relay_smtp_password": isDefault(*cfg.MessageExportSettings.GlobalRelaySettings.SmtpPassword, ""),
 		"is_default_global_relay_email_address": isDefault(*cfg.MessageExportSettings.GlobalRelaySettings.EmailAddress, ""),
-		"global_relay_smtp_server_timeout":      *cfg.EmailSettings.SMTPServerTimeout,
+		"global_relay_smtp_server_timeout":      *cfg.MessageExportSettings.GlobalRelaySettings.SMTPServerTimeout,
```
Comment (isacikgoz) [inline services/telemetry/telemetry.go:771]: I found this one with the tool :) We were using `cfg.EmailSettings.SMTPServerTimeout` for `GlobalRelayMessageExport` but now it has it's own value.

**Comment by nronas** [comment]
Is this preparing the surface before we add the `go vet` rule? I get that we can use a comment line to "skip" the check, but I cannot see the check itself

**Comment by isacikgoz** [comment]
@nronas yes, the vet rule is on the way. Before submitting that one, I wanted to get if the skip comment looks OK or not.

**Review by nronas — APPROVED** [review]
LGTM

File: services/telemetry/telemetry.go (hunk 1 above)
Comment (grundleborg) [reply services/telemetry/telemetry.go:771]: Good catch :)

**Review by grundleborg — COMMENTED** [review]
0/5 on this, but maybe we should say `telemetry: none` instead?

5/5 on this one: we should have a PM review all the fields you're putting this tag on to make sure there aren't any that *should* be included in telemetry but got missed out.

**Review by agnivade — APPROVED** [review]
I'd personally prefer struct tags. But It's a non-blocking concern.

File: model/config.go [hunk 2]
```diff
@@ -265,33 +265,33 @@ var ServerTLSSupportedCiphers = map[string]uint16{
...
 	EnableOAuthServiceProvider                        *bool    `access:"integrations"`
 	EnableIncomingWebhooks                            *bool    `access:"integrations"`
 	EnableOutgoingWebhooks                            *bool    `access:"integrations"`
 	EnableCommands                                    *bool    `access:"integrations"`
-	DEPRECATED_DO_NOT_USE_EnableOnlyAdminIntegrations *bool    `json:"EnableOnlyAdminIntegrations" mapstructure:"EnableOnlyAdminIntegrations"` // This field is deprecated and must not be used.
+	DEPRECATED_DO_NOT_USE_EnableOnlyAdminIntegrations *bool    `json:"EnableOnlyAdminIntegrations" mapstructure:"EnableOnlyAdminIntegrations"` // Deprecated: do not use
 	EnablePostUsernameOverride                        *bool    `access:"integrations"`
 	EnablePostIconOverride                            *bool    `access:"integrations"`
-	EnableLinkPreviews                                *bool    `access:"site"`
+	EnableLinkPreviews                                *bool    `access:"site"` // telemetry: none
```
Comment (wiersgallak) [inline model/config.go:294]: @esethna Should this configuration be included in telemetry?

File: model/config.go [hunk 3]
```diff
@@ -353,12 +353,12 @@ type ServiceSettings struct {
...
-	SplitKey                                          *string `access:"environment,write_restrictable"`
-	FeatureFlagSyncIntervalSeconds                    *int    `access:"environment,write_restrictable"`
-	DebugSplit                                        *bool   `access:"environment,write_restrictable"`
-	ThreadAutoFollow                                  *bool   `access:"experimental"`
+	LocalModeSocketLocation                           *string // telemetry: none
+	EnableAWSMetering                                 *bool   // telemetry: none
+	SplitKey                                          *string `access:"environment,write_restrictable"` // telemetry: none
+	FeatureFlagSyncIntervalSeconds                    *int    `access:"environment,write_restrictable"` // telemetry: none
+	DebugSplit                                        *bool   `access:"environment,write_restrictable"` // telemetry: none
+	ThreadAutoFollow                                  *bool   `access:"experimental"`                   // telemetry: none
```
Comment (wiersgallak) [inline model/config.go:361]: @esethna Should this configuration be included in telemetry?

**Review by grundleborg — APPROVED** [review]
LGTM - just waiting on the PM review of excluded settings now.

**Comment by grundleborg** [comment]
@esethna please take a look at Katie's questions on this PR when you have a moment.

**Comment by esethna** [comment]
@grundleborg yes both those configs should ideally have telemetry

**Comment by grundleborg** [comment]
@esethna thank you, can you pleases specify what telemetry you want for them? E.g. for each value that you'd like telemetry for, would you like a boolean "is default" check, would you like the actual value itself (assuming that wouldn't be sensitive), etc.

**Comment by esethna** [comment]
@grundleborg Thanks, for both `ThreadAutoFollow` and `EnableLinkPreviews` let's just track the the actual value (true/false)

**Review by esethna — APPROVED** [review]

**Review by wiersgallak — APPROVED** [review]

**Pull request merged**